app/models/member.rb

```diff
@@ -51,6 +53,21 @@ def first_name
   def self.active
     Member.where(skip_meetings: false)
   end
+
+  ###########################################################
+  ################ Email Confirmation Part ##################
+  ###########################################################
+  def email_activate
+    self.email_confirmed = true
+    self.confirm_token = nil
+    save!(:validate => false)
+  end
+
+  private
+  
+  def confirmation_token
+    self.confirm_token = SecureRandom.urlsafe_base64.to_s if self.confirm_token.blank?
+  end
```

Comment:
Naming here is all over the place. Either it's `confirm_token` or `confirmation_token` :)